clients.asciidoc

```diff
@@ -310,18 +316,60 @@ $
 
 
-=== Parity's Geth Compatibility Mode
+==== JSON-RPC Interface
+
+Ethereum clients offer an Application Programming Interface (API), offering a set of Remote Procedure Call (RPC) commands, which are encoded as JavaScript Object Notation (JSON). You will see this referred to as the _JSON-RPC API_. Essentially, the JSON-RPC API is an interface that allows us to write programs that use the Ethereum client as a _gateway_ into the Ethereum network and blockchain.
+
+Usually, the RPC interface is offered over as an HTTP service on port +8545+. For security reasons it is restricted, by default, to only accept connections from localhost (the IP address of your own computer which is +127.0.0.1+).
+
+To access the JSON-RPC API, you can use a specialized library, written in the programming language of your choice, which provides "stub" function calls corresponding to each available RPC command. Or, you can manually construct HTTP requests and send/receive JSON encoded requests. You can even use a generic command-line HTTP client, like +curl+, to call the RCP interface. Let's try that:
 
+.Using curl to call the web3_clientVersion function over JSON-RPC
 ----
-$ parity --geth
+$ curl -X POST -H "Content-Type: application/json" --data '{"jsonrpc":"2.0","method":"web3_clientVersion","params":[],"id":1}' http://localhost:8545
+{"jsonrpc":"2.0","id":1,"result":"Geth/v1.8.0-unstable-02aeb3d7/linux-amd64/go1.8.3"}
+----
+
+In this example, we use +curl+ to make an HTTP connection to address +http://localhost:8545+. We are already running +geth+ and it is offering the JSON-RPC API as an HTTP service on port 8545. We instruct +curl+ to use the HTTP +POST+ command, and to identify the content as +Content-Type: application/json+. Finally, we pass a JSON-encoded request as the +data+ component of our HTTP request. Most of our command line is just setting up +curl+ to make the HTTP connection correctly. The interesting part is the actual JSON-RPC command we issue:
+
+----
+{"jsonrpc":"2.0","method":"web3_clientVersion","params":[],"id":1}
+----
+
+This request says that we want to use the +2.0+ version of the JSON-RPC, to call the method +web3_clientVersion+ with no parameters and with network id +1+ (the Ethereum mainnet).
+
+The response we receive is:
+
+----
+{"jsonrpc":"2.0","id":1,"result":"Geth/v1.8.0-unstable-02aeb3d7/linux-amd64/go1.8.3"}
+----
+
+This tells us that the JSON-RPC API is being served by Geth client version 1.8.0.
+
+Let's try something a bit more interesting. In the next example, we ask the JSON-RPC API for the current price of gas in wei:
+
+----
+$ curl -X POST -H "Content-Type: application/json" --data '{"jsonrpc":"2.0","method":"eth_gasPrice","params":[],"id":1}' http://localhost:8545
+
+{"jsonrpc":"2.0","id":1,"result":"0x430e23400"}
 ----
 
-////
-TODO
-////
+The response, +0x430e23400+, tells us that the current gas price on mainnet is 1.8 Gwei (gigawei or billion wei).
```

Comment:
Again here, I disagree with the use of the rpc request object member "id", trying to apply it to the identification of the network.

This "id" member should be used to correlate the context between the request and replay rpc objects.